nirc_ehr/resources/queries/dbo/q_blood.sql

```diff
@@ -1,9 +1,15 @@
 
-SELECT anm.ANIMAL_ID_NUMBER AS Id,
-       CAST(anmEvt.EVENT_DATETIME AS TIMESTAMP) AS bloodDate,
+SELECT anmEvt.ANIMAL_EVENT_ID as objectid,
+       anm.ANIMAL_ID_NUMBER AS Id,
+       CAST(COALESCE (adt.CHANGE_DATETIME,anmEvt.EVENT_DATETIME) AS TIMESTAMP) AS bloodDate,
+       CAST(COALESCE (adt.CHANGE_DATETIME,anmEvt.EVENT_DATETIME) AS TIMESTAMP) AS modified,
        anmEvt.RESULT AS quantity,
-       anmCmt.TEXT AS remark
+       anmCmt.TEXT AS remark,
+       CASE WHEN adt.AUDIT_ID IS NULL THEN 'N'
+        ELSE 'Y' END AS isChanged
 FROM ANIMAL_EVENT anmEvt
 LEFT JOIN ANIMAL anm ON anmEvt.ANIMAL_ID = anm.ANIMAL_ID
 LEFT JOIN ANIMAL_EVENT_COMMENT anmCmt ON anmEvt.ANIMAL_EVENT_ID = anmCmt.ANIMAL_EVENT_ID
+LEFT JOIN AUDIT_TRAIL adt ON anmEvt.ANIMAL_EVENT_ID = substring(adt.PRIMARY_KEY_VALUES, 18) -- ANIMAL_EVENT_ID =xxxxxx
+AND adt.PRIMARY_KEY_VALUES LIKE '%ANIMAL_EVENT_ID%'
```

Comment:
Should we want to filter out delete audit events in this query?

Reply:
I thought about that but the deleted audit events will not have a matching row in ANIMAL_EVENT table as it would have been deleted.

Reply:
OK, thanks for verifying that it actually deletes from the source (some apps don't, they filter against the deletion audit record)